chain/network/src/routing/routing_table_actor.rs

```diff
@@ -306,63 +306,56 @@ impl RoutingTableActor {
         prune_edges_not_reachable_for: Duration,
     ) -> Vec<Edge> {
         let now = Instant::now();
-        let mut oldest_time = now;
+        // Save nodes on disk and remove from memory only if elapsed time from oldest peer
+        // is greater than `SAVE_PEERS_MAX_TIME`
+        if !(force_pruning
+            || (self.peer_last_time_reachable.values())
+                .any(|&last_time| now.saturating_duration_since(last_time) >= SAVE_PEERS_MAX_TIME))
+        {
+            return Vec::new();
+        }
```

Comment:
I wonder if this could be made more readable? I see two problems here: 

* it's just a complex multiline condition
* with this implementation we iterate `peer_last_time_reachable` twice -- once to see if any peer is older than `SAVE_PEERS_MAX_TIME`, and once to collect all peers older than prune_edges_not_reachable_for. I don't worry about this from perf perspective, but I do worry about logical duplication. 

The second is probably OK -- this is essentially a fast-path check, so it is defensible that we don't want to allocate a hash set here. 

Yeah, really struggle to suggest a strictly better alternative here. Maaybe just de-morgan this? 

```rust
        // Fast path: if we saw everyone somewhat recently, don't bother with pruning edges. 
        if !force_pruning
            && self.peer_last_time_reachable.values()
                .all(|&last_time| now.saturating_duration_since(last_time) < SAVE_PEERS_MAX_TIME))
        {
            return Vec::new();
        }
```

Reply:
@matklad I moved the condition to a separate variable, to be more readable.
```rust
    let do_pruning = force_pruning
            || (self.peer_last_time_reachable.values())
                .any(|&last_time| now.saturating_duration_since(last_time) >= SAVE_PEERS_MAX_TIME);
        if !do_pruning {
            return Vec::new();
        }
```